mne/epochs.py

```diff
@@ -178,6 +178,11 @@ def __init__(self, info, data, events, event_id, tmin, tmax,
         del event_id
 
         if events is not None:  # RtEpochs can have events=None
+
+            if not np.array_equal(events, np.array(events, dtype=int)):
+                raise ValueError('events values be integers.')
+            events = np.array(events, dtype=int)
```

Comment:
it's will be not needed. And it's btw not the standard way to change the dtype of an array. 

events = events.astype(int)

is the standard way.